crates/kubelet/src/handle.rs

```diff
@@ -33,27 +34,36 @@ pub trait Stop {
     async fn wait(&mut self) -> anyhow::Result<()>;
 }
 
+/// Trait to describe necessary behavior for creating multiple log readers.
+/// TODO: Both providers make a handle containing a tempfile. If this is a common pattern,
+/// it might make sense to provide that implementation here. This would add `tempfile` as a
+/// dependency of `kubelet`.
+pub trait LogHandle<R>: Sync + Send {
+    /// Create new log reader.
+    fn output(&self) -> R;
```

Comment:
Not sure `output` is the best name. In a way `R` here is really a handle and the type we're calling `LogHandle` is more like a handle dispenser, but I'm not sure how to make that more clear.

Reply:
Could rename to `build` and LogHandle could be LogHandleBuilder.

Reply:
While I'm not sure it's the best name, I do think it's an improvement 😀

Reply:
Went with Factory, builder didn't seem quite right since we aren't implementing the real builder pattern here.